### PR TITLE
Support more than 30 status

### PR DESCRIPTION
**Pull request by Jrc356**
We have a monorepo that currently runs 44 jobs on each PR. This leads to a mismatch between the expected amount of statuses and the current completed amount which makes the gatekeeper pass earlier than expected. The issue is that the status API default size is 30. This PR bumps the `PerPage` size to 100 (max allowed) and adds the necessary page continuation logic to effectively allows for _infinite_ statuses to be retrieved.

### CODE REVIEW

**Comment by rytswd** [comment]
Thank you @Jrc356! I believe this fixes #42. Let us check this further, as we would ideally want to have a test case to ensure this doesn't break the existing code, and also we can handle cases where we have exactly 100, more than 100, and more than 200. As I can see the test cases to be a bit annoying to work out, we may decide to go with this change without such coverage, but would like to get a closer look before deciding to merge this.

**Comment by Jrc356** [comment]
> Thank you @Jrc356! I believe this fixes #42. Let us check this further, as we would ideally want to have a test case to ensure this doesn't break the existing code, and also we can handle cases where we have exactly 100, more than 100, and more than 200. As I can see the test cases to be a bit annoying to work out, we may decide to go with this change without such coverage, but would like to get a closer look before deciding to merge this.

ah, yep, that was the exact issue I came across as well. Lemme know if/how I can help!

**Comment by rytswd** [comment]
I found the implementation easy enough to follow, so just need some test cases to prove it's doing what's expected - if you could help with that, I think we can merge this relatively quickly! 🥰

**Comment by Jrc356** [comment]
@rytswd I've added some test cases, could you take another look? 🙂

File: internal/validators/status/validator_test.go
Comment (rytswd) [inline internal/validators/status/validator_test.go:807]: I cannot remember the exact use case, but I would imagine the pagination support would be needed for `ListCheckRunsForRefFunc` API call as well?

**Review by rytswd — COMMENTED** [review]
Mostly looking good, but left a couple of comments.
Also I'd like to have @hlts2 to have a look as well

File: internal/validators/status/validator.go
Comment (hlts2) [inline internal/validators/status/validator.go:153]: Could you please add also pagination logic to `ListCheckRunsForRef` as well? 🙏 

https://github.com/upsidr/merge-gatekeeper/pull/48/files#diff-5c0d013fe9e031ae272ee0c3d741f7929c2a7aaf9fae9b396d820968c65fb0f7L175

File: internal/validators/status/validator_test.go
Comment (Jrc356) [reply internal/validators/status/validator_test.go]: ah yeah, good catch. Updated this for all the new mock functions in this PR

File: internal/validators/status/validator_test.go
Comment (Jrc356) [reply internal/validators/status/validator_test.go:826]: yep, good call, implemented

File: internal/validators/status/validator.go
Comment (Jrc356) [reply internal/validators/status/validator.go]: done 👍

**Review by rytswd — APPROVED** [review]
Thanks, LGTM!

**Comment by rytswd** [comment]
I'll leave this to @hlts2 to have a final check before merging it 👍

**Comment by Jrc356** [comment]
@hlts2 would you be able to review today or Monday? We'd love to use this but can't without this feature

**Review by hlts2 — APPROVED** [review]
Thanks! LGTM!

**Comment by rytswd** [comment]
@Jrc356 As you noticed, we don't have a set schedule for this. We would at least keep the change in the main branch for some time before cutting a release, so that we can use the latest update in our internal repositories. After confirming there has been no degradation or unintended impact, we release it out as a patch/minor version bump.

As this is rather a critical fix for reliability, we will aim to release it sooner than usual - we should have sufficient runs in a day or two, and would cut a release if no issue found 👍

Having said that, I'm going ahead with the merge for this. Huge thanks for your contribution! 🥰

**Pull request merged**
